posthog/migrations/0302_add_user_pending_email_and_is_verified.py

```diff
@@ -0,0 +1,25 @@
+# Generated by Django 3.2.16 on 2023-02-22 03:55
+
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ("posthog", "0301_organization_enforce_2fa"),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name="user",
+            name="is_email_verified",
+            field=models.BooleanField(blank=True, null=True),
+        ),
+        migrations.AddField(
+            model_name="user",
+            name="pending_email",
+            field=models.EmailField(
+                blank=True, max_length=254, null=True, verbose_name="pending email address awaiting verification"
+            ),
+        ),
```

Comment:
Hm so your proposal is that:
1. When someone signs up, set `email` and `pending_email` to the same value.
2. Once they verify, we clear the `pending_email`. 
3. If someone has `email` but no `pending_email` we assume they are verified.
  - This raises an issue with existing accounts who have not verified. How do we tell the difference between someone who never verified before this feature, and someone who has verified after this feature?
4. If someone has a `pending_email` and an `email`, and they don't match, let them log in with the `email` until they verify the `pending_email`

I feel like the current fields are more explicit and make it easier to handle existing unverified users.

Reply:
> introduces the possibility of there being a record with a non-null `pending_email`, yet `is_email_verified` set to true – which doesn't seem to be desired.

That is because their current email is verified, but they want to switch to a different email address, which requires verification first.

Reply:
That's fair. Well, I think this could be signaled the same with just `pending_email`, but perhaps it's more of a matter of taste, so let's leave the implementation that's already in place